deflate.h

```diff
@@ -14,6 +14,12 @@
 #include "zendian.h"
 #include "crc32.h"
 
+#ifdef S390_DFLTCC_DEFLATE
+#  include "arch/s390/dfltcc_common.h"
+#else
+typedef struct {} arch_deflate_state;
```

Comment:
Empty struct is not allowed in C.

Add
```c++
char empty;
```
as in a similar case:

https://github.com/zlib-ng/zlib-ng/blob/cd721383bf502c76a00b07e465bc1bc8eaa6c5be/cpu_features.h#L23-L37